### PR TITLE
feat: [CHK-1935] elastic dashboard for auth request

### PR DESCRIPTION
**Pull request by pietro-tota**
#### List of Changes
Add span creation for trace all transaction status update made by external entities such as Payment Gateway (PATCH authorization outcome) 
Created spans have the following attributes:

- updateTransactionStatus.type -> Discriminate transaction update state typology (authorization outcome or send payment result)
- updateTransactionStatus.outcome -> Transaction update state outcome: ok, processing error or specific error cause, such as no transaction found for input id, invalid input request, incoherent transaction status and so on
- updateTransactionStatus.trigger -> The external entity which triggered the transaction status update such as Nodo, NPG, PGS (xpay or vpos)

Finally all attribute keys have been generated once in order to avoid overhead parsing it every time, as recommended by open telemetry specifications: 
Note: It is strongly recommended to use [setAttribute(AttributeKey, Object)](https://javadoc.io/static/io.opentelemetry/opentelemetry-api/1.34.0/io/opentelemetry/api/trace/Span.html#setAttribute(io.opentelemetry.api.common.AttributeKey,T)), and pre-allocate your keys, if possible.

#### Motivation and Context
Those modification are needed in order to trace all transaction update state that happens to a transaction in order to properly monitor eCommerce service operatibity

#### How Has This Been Tested?
Junit tests + test in DEV environment (checking span creation)

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/transactions/utils/UpdateTransactionStatusTracerUtils.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    /**
+     * Span attribute used to discriminate transaction update status operation type
+     *
+     * @see UpdateTransactionStatusType
+     */
+    static final AttributeKey<String> UPDATE_TRANSACTION_STATUS_TYPE_ATTRIBUTE_KEY = AttributeKey
+            .stringKey("updateTransactionStatus.type");
+    /**
+     * Spqn attribute used to trace status update outcome
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/utils/UpdateTransactionStatusTracerUtils.java:30]: Typo
```suggestion
     * Span attribute used to trace status update outcome
```

File: src/main/java/it/pagopa/transactions/utils/UpdateTransactionStatusTracerUtils.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+     *
+     * @see UpdateTransactionStatusOutcome
+     */
+    static final AttributeKey<String> UPDATE_TRANSACTION_STATUS_OUTCOME_ATTRIBUTE_KEY = AttributeKey
+            .stringKey("updateTransactionStatus.outcome");
+    /**
+     * Spqn attribute used to trace status update trigger
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/utils/UpdateTransactionStatusTracerUtils.java:37]: Typo
```suggestion
     * Span attribute used to trace status update trigger
```

File: src/main/java/it/pagopa/transactions/utils/UpdateTransactionStatusTracerUtils.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        /**
+         * Transaction status update triggered by PGS (VPOS)
+         */
+        PGS_VPOS,
+
+        /**
+         * Transaction status update triggered by PGS (XPAY)
+         */
+        PGS_XPAY
+    }
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/utils/UpdateTransactionStatusTracerUtils.java:82]: I'm not convinced by this encoding, as it allows invalid combinations (such as `(NPG, SEND_PAYMENT_OUTCOME)`)

Comment (pietro-tota) [reply src/main/java/it/pagopa/transactions/utils/UpdateTransactionStatusTracerUtils.java]: Made span parameter combination encoded into code in order to not allow for invalid combination.
See [this commit](https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/390/commits/360e43bfd75d3c71feb9058497859071365853f5)

**Review by infantesimone — APPROVED** [review]

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**